packages/samples/headless-react/src/pages/DependentFacet.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  buildFacet,
+  buildSearchEngine,
+  Facet as HeadlessFacet,
+  getSampleSearchEngineConfiguration,
+  loadFacetSetActions,
+  SearchEngine,
+} from '@coveo/headless';
+import {Component} from 'react';
+import {Facet} from '../components/facet/facet.fn';
+import {AppContext} from '../context/engine';
+
+export class DependentFacet extends Component<
+  {},
+  {visibleDependantFacet: boolean}
+> {
+  private engine: SearchEngine;
+  private facets: HeadlessFacet[];
+  constructor(props: {}) {
+    super(props);
+    this.engine = buildSearchEngine({
+      configuration: getSampleSearchEngineConfiguration(),
+    });
+    const fields = ['objecttype', 'author', 'source', 'filetype'];
+    this.facets = fields.map((field) =>
+      buildFacet(this.engine, {options: {field}})
+    );
+    this.state = {visibleDependantFacet: false};
+  }
+
+  componentDidMount() {
+    this.engine.executeFirstSearch();
+    this.makeDependent();
+  }
+
+  render() {
+    return (
+      <AppContext.Provider value={{engine: this.engine}}>
+        <div>
+          <h2>Parent facet</h2>
+          <Facet controller={this.parentFacet} />
+        </div>
+        <div>
+          <h2>Dependent facet</h2>
+          <div
+            style={{
+              display: this.state.visibleDependantFacet ? 'block' : 'none',
```

Comment:
Would it be possible to use JSX instead of CSS to show/hide? Or does that interfere with the sub/unsub logic? e.g.

```
this.state.showDependentFacet && <Facet ... />
```